### PR TITLE
fix: follow up page group support for root/body page type and concatenated docs

### PR DESCRIPTION
**Pull request by MurakamiShinyu**
Follow up page-group support for CSS GCPM `:nth(... of <page-type>)` in edge cases found after the initial implementation.

- Handle page type specified on root/body consistently: ensure firstPageType is initialized from root/body page declarations so page-group matching starts correctly from the first page.

- Fix page-group index tracking across concatenated HTML sources: avoid carrying indices between documents by tracking counts per element and handling document boundaries explicitly.

- Prevent synthetic boundary blank pages (issue #666 behavior) from consuming page-group indices: treat inserted blank pages between concatenated documents as non-counting for `:nth(... of <page-type>)` progression.

- Preserve page-group tracking state during target-counter rerender paths to avoid accidental double counting/regression.

- Add regression coverage for concatenated documents with a boundary blank page to verify that the second document restarts page-group counting at `:nth(1 of A)` and still matches Named Page A / Page Group A correctly.

follow-up for issue #1710, PR #1745, and PR #1746

Assisted-by: GitHub Copilot Chat:gpt-5.3-codex

<details>
<summary>How the AI was used</summary>

- After PR #1745 merged, the human author found further edge cases (root/body page-type handling, a forced page-break test variant using left/right) and directed the fix, citing the exact CSS GCPM draft spec wording for `break-after` handling when the AI's first attempt needed correction.
- The human author found a page-group indexing regression when concatenating HTML documents with a synthetic blank page, pointed the AI to the related issue #666 / PR #694 code as a hint, and asked for a regression test once the fix worked — catching that page 5's `:nth(1 of A)` and "Page Group A" header were missing before the fix was complete.
- The human author decided to combine this fix with an earlier not-yet-committed root/body page-type fix into one PR and directed the commit message content; the AI implemented per that direction.

</details>

### CODE REVIEW

File: packages/core/src/vivliostyle/css-styler.ts [hunk 1]
```diff
@@ -517,6 +517,14 @@ export class Styler implements AbstractStyler {
...
     this.offsetMap.addStuckRange(rootOffset);
     const style = this.getAttrStyle(this.root);
+    const rootPageCV = style["page"] as CssCascade.CascadeValue;
+    const rootPageType =
+      rootPageCV &&
+      !Css.isDefaultingValue(rootPageCV.value) &&
+      rootPageCV.value.toString();
+    if (rootPageType && rootPageType.toLowerCase() !== "auto") {
+      this.cascade.firstPageType = rootPageType;
+    }
```
[the remark this answers is not in the document]
Comment (MurakamiShinyu) [reply packages/core/src/vivliostyle/css-styler.ts]: Fixed.
